onnxruntime/test/providers/provider_test_utils.cc

```diff
@@ -34,6 +34,42 @@ void Check(const OpTester::Data& expected_data, const Tensor& output_tensor, con
   }
 }
 
+template <>
+void Check<double>(const OpTester::Data& expected_data, const Tensor& output_tensor, const std::string& provider_type) {
```

Comment:
Need this method to be specialized for double as otherwise it goes through the default version of this templated method which checks for explicit equality of values and fails some tests that got added in the PR.